sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -498,10 +498,8 @@ case class DataSource(
       outputColumnNames: Seq[String],
       physicalPlan: SparkPlan): BaseRelation = {
     val outputColumns = DataWritingCommand.logicalPlanOutputWithNames(data, outputColumnNames)
-    if (outputColumns.map(_.dataType).exists(_.isInstanceOf[CalendarIntervalType])) {
```

Comment:
Let's do this change after we officially make `CalendarIntervalType` public. i.e. move it to a public package.

Reply:
Just wondering what's the relation between this PR and opening `CalendarIntervalType`? An `INTERVAL` column could appear as the result of subtraction of 2 datetime columns, and an user may want to store it into fs.

Reply:
interval type is kind of an interval type for now. It's a big decision if we can read/write it from/to data sources.

Reply:
and Python and R needs a proper conversion for both to read and write as well.

Reply:
How are Python and R involved into read/write in parquet?

Reply:
For instance, if Scala API saves interval types:

```scala
df.save.parquet("")
```

and Python reads it.

```python
spark.read.parquet("...").collect()
```

There's no way to map it in Python side via `collect`. In case of Date type, it's mapped to `date.date` instance in Python.

Reply:
We gotta make it all supported before exposing it all related interval ones (see https://github.com/apache/spark/pull/25022#issuecomment-514472372)